api/base/middleware.py

```diff
@@ -47,3 +51,25 @@ def process_response(self, request, response):
                 raise err
         commands.disconnect()
         return response
+
+
+class FlaskRequestMiddleWare(object):
+    """
+    Push and pop new flask request contexts alongside Django requests
+
+    This is required to prevent caching issues with ModularODM, since (Flask)StoredObject is keyed on the flask
+    request object.
+    """
+    def process_request(self, request):
+        ## Called on every request, so self.flask_ctx should always be defined
+        self.flask_ctx = dummy_app.test_request_context()
+        self.flask_ctx.push()
+
+    def process_exception(self, request, exception):
```

Comment:
As far as I can tell from the code,there seem to be a few edge cases (mostly malformed view functions) where an exception could be raised in ways that would prevent response middleware from firing:

https://github.com/django/django/blob/master/django/core/handlers/base.py#L106